### DISCUSSION TITLE
How to make a time-consuming request not affect other types of requests.

### DISCUSSION THREAD

**Discussion by byrnexu** [uNetworking/uWebSockets#1417] [Q&A]
Let me ask a question. Suppose a websocket request is received, and the server will have a very time-consuming calculation to get the result and return it to the client. Since my uwebsockets server is single-threaded, it will block other requests, which is very time-consuming The calculation of time must be ordered, so I can't use multi-threading. In this case, is there any way to make the processing of this request not affect other types of requests.

**Comment by alexxmanea** [comment]
Did you find a way to do this? I'm also trying to achieve the same thing as you described. Thanks!

**Comment by ghost** [comment]
Look at Loop::defer
